src/test/java/tictactoe/CommandLineTest.java

```diff
@@ -21,30 +21,30 @@ public void initialize() {
     }
 
     @Test
-    public void itWriteInTheCommandLine() throws Exception {
+    public void displayTheInput() throws Exception {
         cli.write("hello");
         assertEquals( "hello", out.toString());
     }
 
     @Test
-    public void itReadFromTheCommandLine() throws Exception {
+    public void readTheUserInput() throws Exception {
         assertEquals(cli.read(), "world");
     }
 
     @Test
-    public void itRaiseAnException() throws Exception {
-        BufferedReader input = new FakeBuffered(new StringReader("world"));
+    public void handleIOException() throws Exception {
+        BufferedReader ioExceptionThrower = new IOExceptionThrower();
         out = new StringWriter();
         PrintWriter output = new PrintWriter(out, true);
-        cli = new CommandLine(input, output);
+        cli = new CommandLine(ioExceptionThrower, output);
 
         assertEquals(cli.read(), "Error");
     }
 }
 
-final class FakeBuffered extends BufferedReader {
-    FakeBuffered(final Reader reader) {
-        super(reader);
+final class IOExceptionThrower extends BufferedReader {
```

Comment:
I don't thinks it's particularly unidiomatic. I'd argue that it is mostly about re-use and legibility. I am surprised your IOexceptionThrower "only" needs and empty reader to fire off an exception. I would have overwritten "read(...)" (or similar) to always throw...